extensions/markdown-language-features/src/test/markdownLink.test.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { checkSmartPaste, createLinkSnippet } from '../languageFeatures/copyFiles/shared';
+import 'mocha';
+import { validateLink } from '../languageFeatures/copyFiles/copyPasteLinks';
+
+suite('uriToWorkspaceEdit', () => {
+
+	// end to end test of checkSmartPaste & createLinkSnippet
+	// check multicursor (end to end)
+
+	// check paste with selection
+	// check paste over markdown link
+	// check paste over markdown image
+	// check paste within markdown link
+	// check paste with open bracket
+
+	// check image link
+
+	////////////////////////////////// validateLink tests //////////////////////////
```

Comment:
To help organize these tests, you can create a `suite` inside the existing one. Just wrap these lines in:

```
suite('validateLink', () => {
    // Tests go here
})
```